src/main/java/io/pravega/connectors/flink/AbstractReaderBuilder.java

```diff
@@ -0,0 +1,161 @@
+        return builder();
+    }
+
+    /**
+     * Add a stream to be read by the source, from the earliest available position in the stream.
+     * @param streamSpec the unqualified or qualified name of the stream.
+     * @return A builder to configure and create a reader.
+     */
+    public B forStream(final String streamSpec) {
+        return forStream(streamSpec, StreamCut.UNBOUNDED);
+    }
+
+    /**
+     * Add a stream to be read by the source, from the given start position in the stream.
+     * @param stream Stream.
+     * @param startStreamCut Start {@link StreamCut}
+     * @return A builder to configure and create a reader.
+     */
+    public B forStream(final Stream stream, final StreamCut startStreamCut) {
+        return forStream(stream, startStreamCut, StreamCut.UNBOUNDED);
+    }
+
+    /**
+     * Add a stream to be read by the source, from the given start position in the stream to the given end position.
+     * @param stream Stream.
+     * @param startStreamCut Start {@link StreamCut}
+     * @param endStreamCut End {@link StreamCut}
+     * @return A builder to configure and create a reader.
+     */
+    public B forStream(final Stream stream, final StreamCut startStreamCut, final StreamCut endStreamCut) {
+        streams.add(StreamSpec.of(stream, startStreamCut, endStreamCut));
+        return builder();
+    }
+
+    /**
+     * Add a stream to be read by the source, from the earliest available position in the stream.
+     * @param stream Stream.
+     * @return A builder to configure and create a reader.
+     */
+    public B forStream(final Stream stream) {
+        return forStream(stream, StreamCut.UNBOUNDED);
+    }
+
+    /**
+     * Gets the Pravega configuration.
+     */
+    protected PravegaConfig getPravegaConfig() {
+        Preconditions.checkState(pravegaConfig != null, "A Pravega configuration must be supplied.");
+        return pravegaConfig;
+    }
+
+    /**
+     * Resolves the streams to be provided to the reader, based on the configured default scope.
+     */
+    protected List<StreamWithBoundaries> resolveStreams() {
+        Preconditions.checkState(!streams.isEmpty(), "At least one stream must be supplied.");
+        PravegaConfig pravegaConfig = getPravegaConfig();
+        return streams.stream()
+                .map(s -> StreamWithBoundaries.of(pravegaConfig.resolve(s.streamSpec), s.from, s.to))
+                .collect(Collectors.toList());
+    }
+
+    protected abstract B builder();
+
+    /**
+     * A Pravega stream with optional boundaries based on stream cuts.
+     */
+    @Data
+    private static class StreamSpec implements Serializable {
+
+        private static final long serialVersionUID = 1L;
+
+        private final String streamSpec;
+        private final StreamCut from;
+        private final StreamCut to;
+
+        public static StreamSpec of(String streamSpec, StreamCut from, StreamCut to) {
```

Comment:
Well, the main idea here is that the 'streamSpec' is an unprocessed user input that will be parsed later (out of necessity).  I avoided 'streamName' because I don't want anyone to treat the value as an actual stream name.  Also it is a private class.